pkg/scheduler/scheduler.go

```diff
@@ -462,10 +475,30 @@ func (m *Scheduler) processDeletePartitionConfigsEvent(event *schedulerevent.Sch
     }
 }
 
+// Add a scheduling node based on the node added to the cache.
+func (m *Scheduler) processNodeEvent(event *schedulerevent.SchedulerNodeEvent) {
+    // process the node addition (one per event)
+    if event.AddedNode != nil {
+        nodeInfo := event.AddedNode.(*cache.NodeInfo)
+        m.clusterSchedulingContext.addSchedulingNode(nodeInfo)
+    }
+    // process the node deletion (one per event)
+    if event.RemovedNode != nil {
+        nodeInfo := event.RemovedNode.(*cache.NodeInfo)
+        m.clusterSchedulingContext.removeSchedulingNode(nodeInfo)
+    }
+    // preempted resources have now been released update the node
+    if event.PreemptedNodeResources != nil {
+        m.clusterSchedulingContext.releasePreemptedResources(event.PreemptedNodeResources)
```

Comment:
When we add them to the node we do that synchronously *within* the scheduler code. When we confirm the allocation it comes out of the cache.
Using a scheduler event inside the scheduler is not done anywhere we can but that would be a departure of what we have now.

Reply:
I see the reason now. in events.go, since this defines public events
```
type SchedulerNodeEvent struct {
    AddedNode              interface{}
    RemovedNode            interface{}
    PreemptedNodeResources []NodeResource
}
```
can we rename `NodeResource` to `NodePreemptedResource` to avoid confusion?

Reply:
I had thought about keeping it generic so we could reuse it later but after thinking about it longer: preemption for now seems to be the only case we need it. I renamed the object